src/ed_object_models/spawn_sdf_in_gazebo.py

```diff
@@ -0,0 +1,85 @@
+#! /usr/bin/python
+
+import rospy
+import yaml
+import os
+import glob
+
+from rospkg import RosPack
+from gazebo_msgs.srv import SpawnModel
+from geometry_msgs.msg import Pose
+
+
+def from_yaml(yaml_file):
+    '''
+    Spawns a list of sdf files from a yaml file into Gazebo.
+    :param yaml_file: path to a yaml file.
+    :type yaml_file: str
+    :return: Spawns the sdf models.
+    '''
+
+    # Initialize ROS node
+    rospy.init_node('gazebo_simulator_object_spawner')
+
+    # Wait until gazebo is ready to spawn sdf models
+    rospy.wait_for_service('gazebo/spawn_sdf_model')
+    spawn_model_prox = rospy.ServiceProxy('gazebo/spawn_sdf_model', SpawnModel)
+
+    # Load yaml with list of objects that need to be loaded
+    yaml_file_path = None
+    if os.path.isfile(yaml_file):   # Check if yaml_file is path to a file
+        yaml_file_path = yaml_file
+    else:   # Check if yaml_file is path to a file in fast_simulator_data
+        package_path = RosPack().get_path('fast_simulator_data')
+        if os.path.isfile(package_path + yaml_file):
+            yaml_file_path = package_path + yaml_file
+
+    if yaml_file_path is None:
+        print('Warning: Could not find yaml file.')
+        return
+
+    with open(yaml_file_path, 'r') as f:
+        data = yaml.safe_load(f)
+
+    # Get paths in $GAZEBO_MODEL_PATH
+    model_paths = os.environ['GAZEBO_MODEL_PATH'].split(os.pathsep)
+
+    # Iterate over objects and spawn
+    for sod in data:
```

Comment:
sod?

Reply:
Janno suggested loading a list of models from a yaml file and wrote some example code, which contained the name sod. What would be a more logical name, item maybe?

Reply:
already much better